api/tsconfig.json

```
@@ -0,0 +1,15 @@
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "declaration": true,
+    "removeComments": true,
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
+    "target": "es2017",
+    "sourceMap": true,
+    "outDir": "./dist",
+    "baseUrl": "./",
+    "incremental": true
```

Running `start`, `build` or `start:dev` are all giving me typescript errors atm, looks like duplicate definitions for standard libraries.

I think the issue here comes from `styled-components` being installed in some of the other packages in the repo, and polluting the shared `node_modules`, but we'll need this here to be able to compile the api now.

cc @comoser 

Adding `skipLibCheck` seems to fix the issue.

```suggestion
    "incremental": true,
    "skipLibCheck": true
```

---

We should be able to specify the `styled-components` dependency only on the web-client, is there a way @s0 ?

---

With `yarn workspaces`, I don't believe that's possible, unless another package in the workspace is using a different version of that package.

I've used [rush](https://rushjs.io/) in other repos alongside pnpm, which has considerably more advanced dependency management, and prevents phantom dependencies. I prefer it and typically use it on my own projects, the downside however is it's a lot more opinionated, and a little bit more fiddly to add / remove packages, and has a bit more of a learning curve compared to yarn workspaces.

---

And of course it's yet another development tool to install...

---

For now, i'd recommend just adding the skip lib check in the projects that need it, and continuing with yarn workspaces.